### PR TITLE
Fix Schema.org handling

### PR DESCRIPTION
**Pull request by patrys**
Re #1349 

Google does not like when a page has more than one main element described (it parses it correctly but will not trigger a rich snippet as it assumes the page to be a carousel of various items). That's why I've dropped the schema generation from product lists: we're sending out data that is not used for anything.

1. Remove `WebPage` from pages other than the home page
2. Remove `Product` schema from pages other than product details
3. Fix `Product` `image` URL and list all images
4. Fix `Product` `offers` to include all variants
5. Drop React component that tries to handle the schema

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: saleor/product/utils.py [hunk 1]
```diff
@@ -109,26 +109,31 @@ def products_for_cart(user):
...
-            data['offers']['price'] = availability.price_range.min_price.net
-
-        if availability.available:
-            data['offers']['availability'] = 'http://schema.org/InStock'
-        else:
-            data['offers']['availability'] = 'http://schema.org/OutOfStock'
+    for variant in product.variants.all():
+        price = variant.get_price_per_item()
+        available = 'http://schema.org/InStock'
+        if not product.is_available() or not variant.is_in_stock():
```
Comment (Pacu2) [inline saleor/product/utils.py]: I think we should only check if `variant.is_in_stock()` here, handling product without variants could happen on different level.

File: saleor/product/utils.py [hunk 2]
```diff
@@ -109,26 +109,31 @@ def products_for_cart(user):
...
+            available = 'http://schema.org/OutOfStock'
+        variant_data = {
```
Comment (Pacu2) [inline saleor/product/utils.py]: We can add `category` here

File: saleor/product/utils.py [hunk 3]
```diff
@@ -109,26 +109,31 @@ def products_for_cart(user):
...
+def product_json_ld(product, attributes=None):
     # type: (saleor.product.models.Product, saleor.product.utils.ProductAvailability, dict) -> dict  # noqa
     """Generates JSON-LD data for product"""
     data = {'@context': 'http://schema.org/',
             '@type': 'Product',
             'name': smart_text(product),
-            'image': smart_text(product.get_first_image()),
+            'image': [
+                product_image.image.url
+                for product_image in product.images.all()],
```
Comment (Pacu2) [inline saleor/product/utils.py]: We could list images as `ImageObject`, providing dimensions and alt text

File: templates/home.html [hunk 4]
```diff
@@ -55,4 +55,5 @@ <h2 class="text-center">{% trans "Featured products" context "Homepage title" %}
       </div>
     </div>
 
+    <script type="application/ld+json">{{ webpage_schema|safe }}</script>
```
Comment (Pacu2) [inline templates/home.html]: I think `webpage_schema` may be moved from context_processor to regular function in such a case

File: saleor/product/utils.py (hunk 1 above)
Comment (patrys) [reply saleor/product/utils.py]: `product.is_available` is for checking the embargo date, products without variants will never enter this `for` loop.

File: saleor/product/utils.py (hunk 2 above)
Comment (patrys) [reply saleor/product/utils.py]: We could but not until we have a single category per product. There's a separate issue for this I believe (making category a foreign key and adding a secondary taxonomy tree).

File: saleor/product/utils.py (hunk 3 above)
Comment (patrys) [reply saleor/product/utils.py]: We could but Google will only ever use the alt text (for image search) and it defaults to the product name.

File: templates/home.html (hunk 4 above)
Comment (patrys) [reply templates/home.html]: It could, yes.

File: saleor/product/utils.py (hunk 3 above)
Comment (Pacu2) [reply saleor/product/utils.py]: Right, this should be easily fetched by the crawler, let's leave it as it is.

**Comment by patrys** [comment]
By the way, I've attempted to add an Organization element with a name and a logo but we don't have a PNG version of the logo suitable for Google.

**Pull request merged**